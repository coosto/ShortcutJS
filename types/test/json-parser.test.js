"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const json_parser_1 = require("../src/json-parser");
const shortcut_1 = require("../src/shortcut");
describe('JsonParser: parse', () => {
    it('throws an error if json is not array', () => {
        expect(() => json_parser_1.JsonParser.parse(shortcut_1.shortcutJS, {})).toThrowErrorMatchingSnapshot();
    });
    it('throws an error if an object is empty', () => {
        expect(() => json_parser_1.JsonParser.parse(shortcut_1.shortcutJS, [{}])).toThrowErrorMatchingSnapshot();
    });
    it('throws an error if combo property is not in an object', () => {
        expect(() => json_parser_1.JsonParser.parse(shortcut_1.shortcutJS, [{ action: 'open' }]))
            .toThrowErrorMatchingSnapshot();
    });
    it('throws an error if action property is not in an object', () => {
        expect(() => json_parser_1.JsonParser.parse(shortcut_1.shortcutJS, [{ combo: 'open' }]))
            .toThrowErrorMatchingSnapshot();
    });
    it('should be ok if is an array of json objects with both properties', () => {
        expect(() => json_parser_1.JsonParser.parse(shortcut_1.shortcutJS, [{ action: 'open', combo: 'ctrl a' }]))
            .not.toThrowError();
    });
});
