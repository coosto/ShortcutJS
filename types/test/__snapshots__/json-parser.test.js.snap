exports[`JsonParser: parse throws an error if action property is not in an object 1`] = `
"
        The json provided must be an array of { combo, action } object. Example
          [
            { action: \'openWindow\', combo: \'ctrl a\' }
          ]
      "
`;

exports[`JsonParser: parse throws an error if an object is empty 1`] = `
"
        The json provided must be an array of { combo, action } object. Example
          [
            { action: \'openWindow\', combo: \'ctrl a\' }
          ]
      "
`;

exports[`JsonParser: parse throws an error if combo property is not in an object 1`] = `
"
        The json provided must be an array of { combo, action } object. Example
          [
            { action: \'openWindow\', combo: \'ctrl a\' }
          ]
      "
`;

exports[`JsonParser: parse throws an error if json is not array 1`] = `"The json provided must be an array"`;
